.github/workflows/test.yml.jinja

```diff
@@ -0,0 +1,65 @@
+name: test Odoo addons
+
+on:
+  pull_request:
+    branches:
+      - "{{ odoo_version }}*"
+  push:
+    branches:
+      - "{{ odoo_version }}*"
+
+{%
+set IMAGES = {
+  "odoo": {
+    "14.0": "ghcr.io/oca/oca-ci/py3.6-odoo14.0:latest",
+    "13.0": "ghcr.io/oca/oca-ci/py3.6-odoo13.0:latest",
+    "12.0": "ghcr.io/oca/oca-ci/py3.6-odoo12.0:latest",
+    "11.0": "ghcr.io/oca/oca-ci/py3.5-odoo11.0:latest",
+    "10.0": "ghcr.io/oca/oca-ci/py2.7-odoo10.0:latest",
+  },
+  "ocb": {
+    "14.0": "ghcr.io/oca/oca-ci/py3.6-ocb14.0:latest",
+    "13.0": "ghcr.io/oca/oca-ci/py3.6-ocb13.0:latest",
+    "12.0": "ghcr.io/oca/oca-ci/py3.6-ocb12.0:latest",
+    "11.0": "ghcr.io/oca/oca-ci/py3.5-ocb11.0:latest",
+    "10.0": "ghcr.io/oca/oca-ci/py2.7-ocb10.0:latest",
+  }
+}
+-%}
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container: {{ "${{" }} matrix.container {{ "}}" }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - container: {{ IMAGES["odoo"][odoo_version] }}
+            makepot: "true"
+          - container: {{ IMAGES["ocb"][odoo_version] }}
```

Comment:
FWIW you can use macros, which can be in a separate file.

As time goes on, there'll be more little stuff that will be different depending on the base odoo version, so maybe having that info in a separate file could help.

Just a suggestion, it's OK as it is anyway. 😊

Reply:
Since this works and is reasonably easy to read I think I'll keep it like this for now. We can refactor later if it becomes too complex.